ipatests/test_integration/test_sudo.py

```diff
@@ -24,22 +24,43 @@
     clear_sssd_cache, get_host_ip_with_hostmask, modify_sssd_conf)
 
 
-class TestSudo(IntegrationTest):
-    """
-    Test Sudo
-    http://www.freeipa.org/page/V4/Sudo_Integration#Test_Plan
-    """
+class TestSudoBase(IntegrationTest):
     num_clients = 1
     topology = 'line'
 
     @classmethod
     def install(cls, mh):
-        super(TestSudo, cls).install(mh)
+        super(TestSudoBase, cls).install(mh)
 
         cls.client = cls.clients[0]
         cls.clientname = cls.client.run_command(
             ['hostname', '-s']).stdout_text.strip()
 
+    def list_sudo_commands(self, user, raiseonerr=False, verbose=False,
+                           password=None):
+        clear_sssd_cache(self.client)
+        sudo_args = '-ll' if verbose else '-l'
+        stdin_text = None
+        if password is None:
+            sudo_args += ' -n'
+        else:
+            sudo_args += ' -S'
+            stdin_text = password + '\n'
+        return self.client.run_command(
+            'su -c "sudo %s" %s' % (sudo_args, user),
```

Comment:
Please pass a list of arguments to ``run_command``. IIRC a string will trigger unsafe ``shell=True`` semantic.